janitor/functions.py

```diff
@@ -3244,7 +3244,7 @@ def groupby_agg(
     :param axis: Split along rows (0) or columns (1).
     :returns: A pandas DataFrame.
     """
-
+    df_c = df.copy()
```

Comment:
I think by convention, we just reassign `df` to `df.copy`:

```suggestion
    df = df.copy()
```